src/main/java/hudson/plugins/analysis/core/AbstractResultAction.java

```diff
@@ -104,12 +106,12 @@ public ToolTipProvider getToolTipProvider() {
      *
      * @return the associated build of this action
      */
-    public final AbstractBuild<?, ?> getOwner() {
+    public final Run<?, ?> getOwner() {
```

Comment:
I have found `@WithBridgeMethods` to be scary in practice, and it seems like overkill here.

Just deprecate this method, returning null or throwing `ClassCastException` if you have to, and introduce a new method with the desired type.